File: app/src/main/java/info/nightscout/androidaps/plugins/pump/medtronic/util/MedtronicConst.java
package info.nightscout.androidaps.plugins.pump.medtronic.util;

import info.nightscout.androidaps.R;

/**
 * Created by andy on 5/12/18.
 */

public class MedtronicConst {

    static final String Prefix = "AAPS.Medtronic.";

    public class Prefs {
        public static final int PumpSerial = R.string.key_medtronic_serial;
        public static final int PumpType = R.string.key_medtronic_pump_type;
        public static final int PumpFrequency = R.string.key_medtronic_frequency;
        public static final int MaxBolus = R.string.key_medtronic_max_bolus;
        public static final int MaxBasal = R.string.key_medtronic_max_basal;
        public static final int BolusDelay = R.string.key_medtronic_bolus_delay;
        public static final int Encoding = R.string.key_medtronic_encoding;
        public static final int BatteryType = R.string.key_medtronic_battery_type;
        public static final int BolusDebugEnabled = R.string.key_medtronic_bolus_debug;
    }

    public class Statistics {

        public static final String StatsPrefix = "medtronic_";
        public static final String FirstPumpStart = Prefix + "first_pump_use";
        public static final String LastGoodPumpCommunicationTime = Prefix + "lastGoodPumpCommunicationTime";
        public static final String LastGoodPumpFrequency = Prefix + "LastGoodPumpFrequency";
        public static final String TBRsSet = StatsPrefix + "tbrs_set";
        public static final String StandardBoluses = StatsPrefix + "std_boluses_delivered";
        public static final String SMBBoluses = StatsPrefix + "smb_boluses_delivered";
        public static final String LastPumpHistoryEntry = StatsPrefix + "pump_history_entry";
        public static final String LastPrime = StatsPrefix + "last_sent_prime";
    }

}

File: medtronic/src/main/java/info/nightscout/androidaps/plugins/pump/medtronic/util/MedtronicConst.kt
package info.nightscout.androidaps.plugins.pump.medtronic.util

import info.nightscout.androidaps.plugins.pump.medtronic.R

/**
 * Created by andy on 5/12/18.
 */
object MedtronicConst {

    const val Prefix = "AAPS.Medtronic."

    object Prefs {
        @JvmField val PumpSerial = R.string.key_medtronic_serial
        @JvmField val PumpType = R.string.key_medtronic_pump_type
        @JvmField val PumpFrequency = R.string.key_medtronic_frequency
        @JvmField val MaxBolus = R.string.key_medtronic_max_bolus
        @JvmField val MaxBasal = R.string.key_medtronic_max_basal
        @JvmField val BolusDelay = R.string.key_medtronic_bolus_delay
        @JvmField val Encoding = R.string.key_medtronic_encoding
        @JvmField val BatteryType = R.string.key_medtronic_battery_type
        val BolusDebugEnabled = R.string.key_medtronic_bolus_debug
    }

    object Statistics {
        const val StatsPrefix = "medtronic_"
        const val FirstPumpStart = Prefix + "first_pump_use"
        const val LastGoodPumpCommunicationTime = Prefix + "lastGoodPumpCommunicationTime"
        const val LastGoodPumpFrequency = Prefix + "LastGoodPumpFrequency"
        const val TBRsSet = StatsPrefix + "tbrs_set"
        const val StandardBoluses = StatsPrefix + "std_boluses_delivered"
        const val SMBBoluses = StatsPrefix + "smb_boluses_delivered"
        const val LastPumpHistoryEntry = StatsPrefix + "pump_history_entry"
        const val LastPrime = StatsPrefix + "last_sent_prime"
        const val LastRewind = StatsPrefix + "last_sent_rewind"
        const val InternalTemporaryDatabase = StatsPrefix + "temporary_entries"
    }
}